Content.Server/Buckle/Components/StrapComponent.cs

```diff
@@ -37,6 +39,50 @@ public class StrapComponent : SharedStrapComponent, IInteractHand, ISerializatio
         [ViewVariables] [DataField("size")] private int _size = 100;
         private int _occupiedSize;
 
+        /// <summary>
+        /// The buckled entity will be offset by this amount from the center of the strap object.
+        /// If this offset it too big, it will be clamped to <see cref="MaxBuckleDistance"/>
+        /// </summary>
+        [DataField("buckleOffset", required: false)]
+        private Vector2 _buckleOffset = Vector2.Zero;
+
+        private bool _enabled = true;
+
+        /// <summary>
+        /// If disabled, nothing can be buckled on this object, and it will unbuckle anything that's already buckled
+        /// </summary>
+        public bool Enabled
+        {
+            get => _enabled;
+            set
+            {
+                _enabled = value;
+                if (_enabled == value) return;
+                RemoveAll();
+            }
+        }
```

Comment:
The buckle verb should check whether the component is enabled. Currently a folded up roller bed has a "buckle yourself" verb in the context menu when it really shouldn't be there.